### DISCUSSION TITLE
[Contribution] How to test packages

### DISCUSSION THREAD

**Discussion by draylegend** [phero-hq/phero#106] [Q&A] [answered]
I forked and cloned the repo, changed source files and built affected packages. How can I test? Do I need to publish with `verdaccio` locally and then create new project with phero cli?

**Comment by JasperH8g** [comment, marked as answer]
Great to read you're doing this! What we do ourselves is differently between `npx phero` and `@phero/server`/`@phero/client`. 

- `npx phero`: Run `npx lerna run build` in the phero-codebase, and in your example-project run `npm i -g ../../path-to-phero-codebase/packages/cli` so you can run `npx phero` to use your local phero. Don't forget to `npm rm -g phero` when you want to use the published version, because `npx` will try locally installed packages first before fetching remote ones.

- `@phero/client`/`@phero/server`: Run `npx lerna run build` in the phero-codebase, and run `npm rm @phero/client && npm i ../../path-to-phero-codebase/packages/client` (and replace `client` with `server` respectively). The uninstalling of the package helps with permissions-issues with the executable in `node_modules/.bin`. 

Does that answer your question?
